### PR TITLE
Add some delight to the Popout component

### PR DESCRIPTION
**Pull request by jossmac**
- Fancy rounded arrow
- More "pop" on the animation

popout-delight

### CODE REVIEW

**Comment by gautamsi** [comment]
Is the animation configurable, would I be able to turn off the animations from config?

File: packages/arch/packages/popout/src/index.js [hunk 1]
```diff
@@ -127,8 +133,9 @@ class Popout extends Component<Props, State> {
   render() {
     const { children, getModalRef, style, width } = this.props;
     let { leftOffset, topOffset, arrowLeftOffset } = this.state;
+    const attachTo = ((document.body: any): HTMLElement); // NOTE: flow junk
 
-    return document.body
+    return attachTo
       ? createPortal(
```
Comment (mitchellhamilton) [inline packages/arch/packages/popout/src/index.js]: Couldn't you remove the `attachTo ? createPortal(...) : null` and just make it `createPortal(...)`?

File: packages/arch/packages/modal-utils/src/transitions.js [hunk 2]
```diff
@@ -197,3 +197,39 @@ export const ZoomInDown = (props: TransitionProps) => {
...
+
+export const SpringDown = (props: TransitionProps) => {
+  const base = {
+    transformOrigin: 'top',
+    transitionProperty: 'opacity, transform',
+    transitionDuration,
+    transitionTimingFunction: 'cubic-bezier(0.2, 0, 0.16, 1.6)',
+  };
+  return (
+    <TransitionReducer
```
Comment (mitchellhamilton) [inline packages/arch/packages/modal-utils/src/transitions.js]: Definitely not all something that we should do in this PR or necessarily soon but I'd love to remove this `TransitionReducer` component at some point and have the api for a transition be `transitionState => styles`

File: packages/arch/packages/popout/src/index.js (hunk 1 above)
Comment (jossmac) [reply packages/arch/packages/popout/src/index.js]: Flow still gets cranky...

File: packages/arch/packages/modal-utils/src/transitions.js (hunk 2 above)
Comment (jossmac) [reply packages/arch/packages/modal-utils/src/transitions.js]: God yes. TBH, most of Arch needs an overhaul.

**Comment by jossmac** [comment]
> Is the animation configurable, would I be able to turn off the animations from config?

@gautamsi No. But you couldn't before either...

**Comment by gautamsi** [comment]
@jossmac I have opened #1037, once that get any traction we should have some option later.

**Pull request merged**